### PR TITLE
190 debugtest updateimage function across all apis

### PR DESCRIPTION
**Pull request by JamesVetro**
Edited each stack to keep images from past versions of the item, and not wipe them, as well as re-adding the base "empty" image files.

### CODE REVIEW

File: server/controller/apis/goalController.js [hunk 1]
```diff
@@ -203,8 +201,9 @@ exports.saveGoal = async ( req, res, redirect ) =>{
         }
         else if ( !req.files || Object.keys( req.files ).length === 0 ) {
             // no files uploaded
-            this.saveGoalImage( req, res, goal.id, 'peak.svg' );
-            
+            if (!goal.goalImage){
```
Comment (briangormanly) [inline server/controller/apis/goalController.js:204]: Why is this a nested if? (and what would happen if goal.goalImage exists?)
Could this just be part of the else if it is nested in?

Also I have a feeling the if statement above:
if ( req.body.goalModified && req.body.goalModified != "false" && !req.files ) {
might have been intended for the same purpose? Do we need both?

**Review by briangormanly — COMMENTED** [review]
See my comments in the review.. if this is the best / intended path I will approve.

File: server/controller/apis/goalController.js (hunk 1 above)
Comment (cmarcusdimarco) [reply server/controller/apis/goalController.js:204]: if ( req.body.goalModified && req.body.goalModified != "false" && !req.files ) {

checks for a field (goalModified) sent from client/agora/views/dashboard/partials/goal/edit-goal.ejs, but goalModified doesn't exist in the back end. I think that's why this if/else has a logic error on back-end API calls.

Testing in the front end also reveals that the image gets deleted if the goal is updated without providing a new image. I've saved the log statements from the console and am setting a Calendly meeting tomorrow to discuss a best approach.

**Comment by Mike-Volchko** [comment]
I'm going to dive into this issue and possibly make some changes.

**Comment by Mike-Volchko** [comment]
The issue with the previous construction of the saveImage call from within save<API> is that it had no handling for when an existing entity was not handed a new image when updating. This is now handled.

**Comment by cmarcusdimarco** [comment]
After meeting with @briangormanly yesterday morning, we discussed the following:

- We should condense the logical flow of the if/elseif/else in the goalController's saveGoal function so that there are only as many conditionals as there are paths. For example, if two different if statements result in the same function call, these should be condensed into a single if statement using logical or.
- A thorough understanding of the saveGoal function and each place it touches is necessary in order to determine the best approach here. The first if condition of saveGoal looks for a field passed from the front end .ejs file, which is why it fails for the back end API call.
- The solution in place, checking for !goal.goalImage, feels like it may be producing the correct results for a reason other than what we intend.

For these reasons, we should discuss this in class on Monday to determine a best way forward, as this function was not written with the intent to update a goal's image from the back end API. Therefore, it becomes a question of API design; do we refactor the back-end update into a separate function, or do we update the conditionals to check for front-end and back-end cases?

In short, let's review the existing code, get familiar with the paths, and discuss Monday.

**Review by cmarcusdimarco — COMMENTED** [review]
After meeting with @briangormanly yesterday morning, we discussed the following:

- We should condense the logical flow of the if/elseif/else in the goalController's saveGoal function so that there are only as many conditionals as there are paths. For example, if two different if statements result in the same function call, these should be condensed into a single if statement using logical or.
- A thorough understanding of the saveGoal function and each place it touches is necessary in order to determine the best approach here. The first if condition of saveGoal looks for a field passed from the front end .ejs file, which is why it fails for the back end API call.
- The solution in place, checking for !goal.goalImage, feels like it may be producing the correct results for a reason other than what we intend.

For these reasons, we should discuss this in class on Monday to determine a best way forward, as this function was not written with the intent to update a goal's image from the back end API. Therefore, it becomes a question of API design; do we refactor the back-end update into a separate function, or do we update the conditionals to check for front-end and back-end cases?

In short, let's review the existing code, get familiar with the paths, and discuss Monday.

**Comment by Mike-Volchko** [comment]
After our discussion post class on Monday, we discussed the three cases when saving an image while updating an entity (New entity with no image passed, existing entity with no image passed, existing/new entity with an image passed). The error within our backend code was located within the most common case (Existing entity with no image passed). This would cause the image to be deleted and replaced with the default image. I noticed a weird redundancy in the if statement for this case which caused it to never be true. I have decided to close this PR out and create a new one with this simple change. 
This change works perfectly through the front end, but we have no idea how to test it through the backend due to the <entity>Modified field being located in the front end. If anyone has an idea of how to do this using Thunder Client it would be appreciated if you could share it amongst the team in the discord. I believe this is a good stopping point for this issue in Milestone 2. We will most likely have to revisit saving images once we decided on how to save an image using a pure API call.
